ci/migrate.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"context"
+
+	"github.com/openmeterio/openmeter/ci/internal/dagger"
+	"github.com/sourcegraph/conc/pool"
+)
+
+func (m *Ci) Migrate() *Migrate {
+	return &Migrate{
+		Source: m.Source,
+	}
+}
+
+type Migrate struct {
+	Source *dagger.Directory
+}
+
+func (m *Migrate) Check(
+	ctx context.Context,
+) error {
+	nix := nix(m.Source)
+	p := pool.New().WithErrors().WithContext(ctx)
+
+	// Always validate schema is generated
+	p.Go(syncFunc(nix.
+		WithExec([]string{"sh", "-c", "nix develop --impure .#dagger -c go generate ./internal/ent/..."}).
```

Comment:
You could use the go moule to generate the code and then extract the directory using `.Container().Directory()`.